lib/req.ex

```diff
@@ -10,92 +10,314 @@ defmodule Req do
 
   @type method() :: :get | :post | :put | :delete
 
+  @default_options %{
+    # request steps
+    auth: nil,
+    base_url: nil,
+    cache: nil,
+    cache_dir: nil,
+    params: [],
+    range: nil,
+    finch: Req.Finch,
+    pool_timeout: 5000,
+    receive_timeout: 15000,
+
+    # response steps
+    follow_redirects: true,
+    location_trusted: false,
+    raw: false,
+
+    # error steps
+    max_retries: 2,
+    retry: true,
+    retry_delay: 2000
+  }
```

Comment:
Since we include all the steps, why can't this be empty and make each step get its options with the corresponding defaults?

Reply:
Good question! I think that's actually preferable but by supplying defaults we can use guards: https://github.com/wojtekmach/req/pull/88/files#diff-07dc6fd4ec95415d610a3e3ddf9374540d6d64d02eaf42df7a233d6f05b32cc1R425. But it's probably not worth it just for guards sake, dunno. Will think about it some more.

Reply:
oops, link doesn't work, here's an example:

```elixir
  def decompress({request, response}) when request.options.raw == true do
     {request, response}
   end
```

Reply:
If we dropped the `raw: false` default the same guard would work though, no?

Reply:
oh, guards shouldn't fail if `options` doesn't have `:raw` key. Yup, thanks!